tests/test_proxy_functional.py

```diff
@@ -113,32 +114,17 @@ async def handler(*args, **kwargs):
     )
 
 
-@pytest.fixture
-def _pretend_asyncio_supports_tls_in_tls(
-    monkeypatch,
-    web_server_endpoint_type,
-):
-    if web_server_endpoint_type != "https" or ASYNCIO_SUPPORTS_TLS_IN_TLS:
-        return
-
-    # for https://github.com/python/cpython/pull/28073
-    # and https://bugs.python.org/issue37179
-    monkeypatch.setattr(
-        asyncio.sslproto._SSLProtocolTransport,
-        "_start_tls_compatible",
-        True,
-        raising=False,
-    )
-
-
-@secure_proxy_xfail(raises=ClientProxyConnectionError)
+@pytest.mark.skipif(
+    not ASYNCIO_SUPPORTS_TLS_IN_TLS,
+    reason="asyncio on this python does not support TLS in TLS",
+)
 @pytest.mark.parametrize("web_server_endpoint_type", ("http", "https"))
-@pytest.mark.usefixtures("_pretend_asyncio_supports_tls_in_tls", "loop")
+@pytest.mark.filterwarnings(r"ignore:.*ssl.OP_NO_SSL*")
```

Comment:
The test was failing because ubuntu threw this warning which caused it to raise an exception and the proxy died

Reply:
Interesting, did you find the cause of the warning? (wondering if we can fix it, or remove the filter in a later release)

Reply:
Yes. document inline and above now

Reply:
Ah, cool, so maybe we can get it fixed in proxy.py then.